orttraining/orttraining/training_ops/cuda/tensor/gather_grad_impl.cu

```diff
@@ -547,6 +670,26 @@ void GatherGradImpl(
       dX_data);
 }
 
+#define PRECOMPUTESPECIALIZED(T, TIndex)                            \
```

Comment:
if this GatherGradImpl is doing something different (i.e., I assume it's precomputing), maybe rename it for clarity?